widget/widgetAssets/js/soarConfigService.service.js

```diff
@@ -0,0 +1,154 @@
+
+    var service = {
+      constants: constants,
+      getKeyStoreRecord: getKeyStoreRecord,
+      executePlaybook: executePlaybook,
+      updateKeyStoreRecord: updateKeyStoreRecord
+    }
+    return service;
+
+    function constants() {
+      return {
+        queryForKeyStore: {
+          "sort": [],
+          "limit": 30,
+          "logic": "AND",
+          "filters": [
+            {
+              "type": "array",
+              "field": "recordTags",
+              "value": [
+                "/api/3/tags/excludeListIndicators"
+              ],
+              "module": "recordTags",
+              "display": null,
+              "operator": "in",
+              "template": "tags",
+              "enableJinja": true,
+              "OPERATOR_KEY": "$",
+              "useInOperator": true,
+              "previousOperator": "in",
+              "previousTemplate": "tags"
+            }
+          ],
+          "__selectFields": [
+            "key",
+            "jSONValue"
+          ]
+        }
+      }
+    }
+
+    function getKeyStoreRecord(queryObject, module) {
+      var defer = $q.defer();
+      var url = API.QUERY + module;
+      $resource(url).save(queryObject, function (response) {
+        defer.resolve(response);
+      }, function (err) {
+        defer.reject(err);
+      })
+      return defer.promise;
+    }
+
+    function updateKeyStoreRecord(keyStoreValue, recordUUID) {
+      $resource(API.API_3_BASE + 'keys' + '/' + recordUUID, null, {
+        'update': {
+          method: 'PUT'
+        }
+      }).update({ 'jSONValue': keyStoreValue}).$promise.then(function () {
+      });
+  }
+
+    function poll(_reqConfig, callback) {
+      return $interval(function () {
+        if (_reqConfig.hasValueReturned) { //check flag before start new call
+          callback(_reqConfig);
+        }
+        _reqConfig.thresholdValue = _reqConfig.thresholdValue - 1; //Decrease threshold value
+        if (_reqConfig.thresholdValue === 0) {
+          stopPoll(_reqConfig); // Stop $interval if it reaches to threshold
+        }
+      }, _reqConfig.pollInterval);
+    }
+    function stopPoll(_reqConfig) {
+      $interval.cancel(_reqConfig.pollPromise);
+      _reqConfig.pollPromise = undefined;
+      _reqConfig.thresholdValue = 0; //reset all flags.
+      _reqConfig.hasValueReturned = true;
+    }
+
+
+    function checkPlaybookExecutionCompletionByPolling(params) {
+      var _pollConfig = {
+        hasValueReturned: true,
+        thresholdValue: 150,
+        pollInterval: params.pollInterval || 2000,
+        pollPromise: undefined,
+        defer: undefined
+      };
+      _pollConfig.defer = $q.defer();
+      _pollConfig.pollPromise = poll(_pollConfig, function (callbackParam) {
+        callbackParam.hasValueReturned = false;
+        $resource(API.WORKFLOW + 'api/workflows/log_list/?format=json&parent__isnull=True&task_id=' + params.taskId).save({},
```

Comment:
use const or define one for URI path like api/workflows/log_list/

Reply:
DONE